src/openms/source/ANALYSIS/MRM/ReactionMonitoringTransition.cpp

```diff
@@ -205,6 +205,22 @@ namespace OpenMS
     }
   }
 
+  int ReactionMonitoringTransition::getProductChargeState() const
+  { 
+    try
+    { 
+      if (!product_.getChargeState() > 0)
+      { 
+        throw Exception::IllegalArgument(__FILE__, __LINE__, __PRETTY_FUNCTION__, "Product charge has never been set");
```

Comment:
this looks a little weird.
Throwing an exception only to catch it again and re-throw it?!